src/synthesiser/Synthesiser.h

```diff
@@ -95,6 +101,19 @@ class Synthesiser {
         return it->second;
     }
 
+    /** Lookup symbol index */
+    size_t lookupSymbolIdx(const std::string& symbol) {
```

Comment:
```suggestion
    std::size_t lookupSymbolIdx(const std::string& symbol) {
```

Design question - should it be `const`?  Name seems to suggest this is a "lookup".

Reply:
It has a side-effect; if the symbol is not found, a new entry will be generated. Perhaps we change it to convert, i.e., converting a symbol to a number.

Reply:
Sorry for not being clear.  I was suggesting that perhaps the name "lookup" implies immutability.  Since the operation always succeeds, perhaps it could be const and the internal map could be mutable.  The interface seems to be "logically" const.
Anyway, it's a minor nitpick.

Reply:
I see what you mean. I like semantically the word "convert" better because it converts a symbol to an index. In old compiler literature, "lookup" is quite often used for this functionality but conversion is more suitable. 

Indeed, we can mark it as const and make the data-structure mutable.